docs/source/notes/designing_new_logic.rst

```diff
@@ -0,0 +1,351 @@
+and the Inform7 Language (I7L). TextWorld's engine interprets TWLL while Inform7 compiles I7L into either
+Z-Machine (.z8) or Glulx (.ulx) games.
+Any additions or modifications to the logic should be defined in **both** languages.
+
+This tutorial goes over the steps needed to add a new type of entity - a push button - in TextWorld.
+
+A. Declaring a new type
+-----------------------
+Each entity within a TextWorld environment has a *type*. A *type* defines the possible interactions the
+player can have with different objects (e.g., open door), as well as their states (e.g. open/closed or
+unlock/locked door).
+Each type must be represented by a unique identifier composed of letters and dashes.
+
+Let's define the a push button type as follows
+
+.. code-block:: bash
+
+    type push-button {
+    }
+
+[TODO: show all basic types and their hierarchy]
+
+NB: notice how `INVENTORY` is all uppercase. This is used to indicate that there should only be one
+object of that type per game.
+
+A *type* can be a subtype (e.g. `food` is a subtype of `portable_object`) which means it will inherit all
+of its parent behaviors (e.g. `food` is also *portable*) and can also define new ones (e.g. `food` is edible).
+
+Let's our push button be a subtype of `thing`, so we can interact with it in a game:
+.. code-block:: bash
+
+    type push-button : thing {
+    }
+
+
+NB: the type `thing` represents everything that is tangible in a game (e.g. `food`, `containers, etc.)
+opposed to abstract concept like `rooms` and `INVENTORY`.
+
+B. Defining a type
+------------------
+
+After declaring the type of an element, its definition needs to be put between curly braces `{...}`. Its
+definition includes various sections.
+
+So far, the `push-button` type will only be recognized by the TextWorld Engine but not the Inform7 compiler.
+To fix that, a mapping between TWLL's type and I7L's counterpart - a `kind` - must be establish.
+
+Let's add the relevant I7L code for the `push-button`:
+.. code-block:: bash
+
+    type push-button : thing {
+
+        inform7 {
+            type {
+                kind :: "push-button";
+            }
+        }
+    }
+
+The above example presents the type in both languages. The designer, here, prefers to add two features to the "door
+type", as openable and lockable. Any feature which is supposed to be as general feature of this type can be mentioned
+in `definition`. Moreover, if there is an initial state that the designer prefers to define to start the game with,
+should be mentioned in the definition part. For instance, the designer of our example game likes to have all doors in
+the game locked by default, when the game starts. Then "A door is usually locked." is added to definition. Similarly
+for a cloth, we can say "cloth are cleanable. A cloth is usually clean.".
+
+Each game usually include a few rooms with various elements in them. The combination of the initial state of each
```

Comment:
Marc, I understand your point; however, I think this description here is helpful for readers to understand the lines of logic file. Since we have planned to separate theses 3 chapters to 3 different file, I can get back to this point later and elaborate this more in the "Game Design Chapter" as well.

Reply:
Maybe chapter 1 should be about "designing a new game"  which would refer to the other chapters.